ruma-api-macros/src/api/request.rs

```diff
@@ -68,13 +86,20 @@ impl Request {
             let field_name = &field.ident;
             let header_name_string = header_name.to_string();
 
-            quote! {
-                #field_name: match headers
-                    .get(#import_path::exports::http::header::#header_name)
-                    .and_then(|v| v.to_str().ok()) // FIXME: Should have a distinct error message
+            let (some_case, none_case) = match &field.ty {
+                syn::Type::Path(syn::TypePath { path: syn::Path { segments, .. }, .. })
+                    if segments.iter().any(|seg| seg.ident == "Option") =>
                 {
-                    Some(header) => header.to_owned(),
-                    None => {
+                    (
+                        quote! {
+                            Some(header.to_owned()),
+                        },
+                        quote! { None, },
```

Comment:
Why is the first `quote!`s content on its own line, the second one's not? I think maybe the whole tuple even fits on the one line in a readable way.